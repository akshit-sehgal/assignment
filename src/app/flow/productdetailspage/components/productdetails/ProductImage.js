import React from 'react';
import PropTypes from 'prop-types';
const ProductImage = (props) => {
    const { imgSrc } = props;
    return (
        <img
            className="pd__image"
            src={imgSrc}
            alt="Product Image"
        />
    );
}
ProductImage.propTypes = {
    imgSrc: PropTypes.string.isRequired
}
export default ProductImage;