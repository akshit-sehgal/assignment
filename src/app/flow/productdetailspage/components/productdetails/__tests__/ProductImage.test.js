import React from 'react';
import { shallow } from 'enzyme';
import ProductImage from '../ProductImage';
import { mockProductDetails } from '../__mocks__/productDetails.mock';

describe('Testing <ProductImage />', () => {
    let props;
    beforeEach(() => {
        props = {
            imgSrc: mockProductDetails.img
        }
    })
    it('should render the default view', () => {
        const wrapper = shallow(<ProductImage {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.pd__image').prop('src')).toEqual(props.imgSrc);
    });

});