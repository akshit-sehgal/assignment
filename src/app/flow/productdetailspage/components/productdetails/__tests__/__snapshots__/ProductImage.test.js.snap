// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProductImage /> should render the default view 1`] = `
<img
  alt="Product Image"
  className="pd__image"
  src="https://images.google.com/1.png"
/>
`;
